internal/server/instance/drivers/driver_qemu.go

```diff
@@ -3473,17 +3514,22 @@ func (d *qemu) isWindows() bool {
 	return strings.Contains(strings.ToLower(d.expandedConfig["image.os"]), "windows")
 }
 
-// isLinux returns whether the VM is Linux. This is the default if no `image.os` is provided.
-func (d *qemu) isLinux() bool {
+// isDarwin returns whether the VM is Darwin.
+func (d *qemu) isDarwin() bool {
 	imageOS := strings.ToLower(d.expandedConfig["image.os"])
-	keywords := []string{"darwin", "mac os", "macos", "windows"}
+	keywords := []string{"darwin", "mac os", "macos"}
 	for _, keyword := range keywords {
 		if strings.Contains(imageOS, keyword) {
-			return false
+			return true
 		}
 	}
 
-	return true
+	return false
+}
+
+// isLinux returns whether the VM is Linux. This is the default if no `image.os` is provided.
+func (d *qemu) isLinux() bool {
```

Comment:
I wonder if it would make sense to have something more format about the Guest OS.

Basically a `inst.GuestOS()` function in `driver_common.go` which does the `image.os` parsing and gives us one of:
- `linux`
- `macos`
- `windows`
- `unknown`

So we can then rely on that everywhere we need to figure out what the OS is both here and in the devices.

Reply:
Except in the driver’s logic, `linux` and `unknown` are the same thing, and it’s not reasonable to keep a list of distros. If you decide to package Rebecca Black OS on the image server, we’d have to also patch Incus to include it, which is not really fun for the one having to maintain this mess.